src/tests/ftest/control/config_generate_run.py

```diff
@@ -44,15 +48,17 @@ def test_config_generate_run(self):
             access_points=server_host, num_engines=num_engines,
             min_ssds=min_ssds, net_class=net_class)
 
-        ap_count = len(server_host.split(","))
         # If there's only one AP, the first line would be a warning message.
         # If we include this message, the server would fail to start, so remove it.
+        ap_count = len(server_host.split(","))
         if ap_count == 1:
             self.log.info("result.stdout (before) = %s", result.stdout)
             result_lines = result.stdout.decode("utf-8").split("\n")
-            result_lines = result_lines[1:]
-            result.stdout = "\n".join(result_lines).encode("utf-8")
-            self.log.info("result (after) = %s", result)
+            # Check that the warning message is the AP message that we're expecting.
+            if result_lines[0] == AP_WARN_MSG:
+                result_lines = result_lines[1:]
+                result.stdout = "\n".join(result_lines).encode("utf-8")
+                self.log.info("result (after) = %s", result)
```

Comment:
Just FYI that something like this is also possible, to avoid splitting and to handle the error message being on ANY line.
```
tmp = result.stdout.decode("utf-8").replace(AP_WARN_MSG + '\n', '')
result.stdout = tmp.encode("utf-8")
```

Reply:
I think this is cleaner. Thanks.